modules/oe_subscriptions_anonymous/tests/src/Functional/SubscribeTest.php

```diff
@@ -209,6 +210,83 @@ public function testForm(): void {
     $this->assertTrue($pages_flag->isFlagged($page_two, $account));
   }
 
+  /**
+   * Tests a case where an email is already taken when a request is submitted.
+   */
+  public function testEmailTakenOnSubscribeRequest(): void {
+    // Create flag and page.
+    $pages_flag = $this->createFlagFromArray([
+      'id' => 'subscribe_page',
+      'entity_type' => 'node',
+      'bundles' => ['page'],
+    ]);
+    $entity = $this->drupalCreateNode([
+      'type' => 'page',
+      'status' => 1,
+    ]);
+
+    // Create a regular user account.
+    $user = $this->createUser(values: ['mail' => 'conflict@example.com']);
+    $this->assertInstanceOf(DecoupledAuthUserInterface::class, $user);
+    $this->assertTrue($user->isCoupled());
```

Comment:
We could use `/** @var .. */` doc instead of the assert.
The goal here is to let the static analysis (phpstan, IDE) know about the type, so it does not complain about the `->isCoupled()` call.

Reply:
Doc is ok.